src/Shells/StrixMusic.Shells.ZuneDesktop/Controls/Views/Collection/ZuneTrackCollection.cs

```diff
@@ -16,154 +20,182 @@ namespace StrixMusic.Shells.ZuneDesktop.Controls.Views.Collection
     /// <summary>
     /// Zune implmenation for the <see cref="ZuneTrackCollection"/>.
     /// </summary>
-    public class ZuneTrackCollection : TrackCollection
+    [INotifyPropertyChanged]
+    public partial class ZuneTrackCollection : CollectionControl<ZuneTrackCollectionItem, ZuneTrackItem>
     {
+        private ObservableCollection<ZuneTrackCollectionItem> _trackItems = new();
+
+        public ZuneTrackCollection()
+        {
+            TrackItems = new ReadOnlyObservableCollection<ZuneTrackCollectionItem>(_trackItems);
+
+            Loaded += ZuneTrackCollection_Loaded;
+            Unloaded += ZuneTrackCollection_Unloaded;
+        }
+
+        private void AttachEvents(ZuneTrackCollectionItem item)
+        {
+            item.PropertyChanged += Item_PropertyChanged;
+        }
+
+        private void DetachEvents(ZuneTrackCollectionItem item)
+        {
+            item.PropertyChanged -= Item_PropertyChanged;
+        }
+
         /// <summary>
-        /// Holds the instance of a artist column textblock.
+        /// The backing dependency property for <see cref="Collection" />.
         /// </summary>
-        public TextBlock? PART_ArtistColumn { get; private set; }
+        public static readonly DependencyProperty CollectionProperty =
+            DependencyProperty.Register(nameof(Collection), typeof(ITrackCollectionViewModel), typeof(ZuneTrackCollection), new PropertyMetadata(null, (d, e) => ((ZuneTrackCollection)d).OnCollectionChanged((ITrackCollectionViewModel?)e.OldValue, (ITrackCollectionViewModel?)e.NewValue)));
```

Comment:
You shouldn't need this CollectionProperty if its present in the base class, simply override the virtual callback.

Reply:
It's not present on the base class. The base class changed in this PR because the item data type changed.